modules/cudf/src/series.ts

```diff
@@ -605,6 +605,23 @@ export class AbstractSeries<T extends DataType = any> {
     return Series.new(this._col.replaceNulls(false, memoryResource));
   }
 
+  /**
+   * Returns a new series with reversed elements.
+   *
+   * @example
+   * ```typescript
+   * import {Series} from '@rapidsai/cudf';
+   *
+   * // Float64Series
+   * Series.new([1, 2, 3]).reverse() // [3, 2, 1]
+   * // StringSeries
+   * Series.new(["foo", "bar"]).reverse() // ["bar", "foo"]
+   * // Bool8Series
+   * Series.new([false, true]).reverse() // [true, false]
+   * ```
+   */
+  reverse(): Series<T> { return Series.new({type: this.type, data: [...this].reverse()}); }
```

Comment:
This will cause a copy to host, won't it? Maybe instead we could construct a reverse sequence and then gather? I think that is what Python cudf does:

https://github.com/rapidsai/cudf/blob/5b8895dc50a5daf99bafede67617e8ebd39d3af6/python/cudf/cudf/core/series.py#L3701-L3704

Reply:
spoke with Bryan a bit, cleared up allocating host memory vs. letting `CUDA/CUDF` do the work. Fixed.